In [26]:
#import libraries
import pandas as pd
import plotly.express as px

In [27]:
#load dataset

In [28]:
dataset=pd.read_csv("preprocessed_data.csv")
dataset

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0         100083       0         Cash loans           M            Y   
1         100145       0         Cash loans           F            Y   
2         100179       0         Cash loans           F            Y   
3         100190       0         Cash loans           M            Y   
4         100295       1         Cash loans           M            Y   
...          ...     ...                ...         ...          ...   
8597      456083       0         Cash loans           F            Y   
8598      456084       0         Cash loans           F            Y   
8599      456140       1         Cash loans           F            Y   
8600      456195       0         Cash loans           F            Y   
8601      456226       0         Cash loans           F            Y   

     FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                  Y             0          103500.0    573628.5      24435.0   
1                  Y             1          202500.0    260725.5      16789.5   
2                  N             0          202500.0    675000.0      53329.5   
3                  N             0          162000.0    263686.5      24781.5   
4                  N             1          225000.0   1019205.0      31032.0   
...              ...           ...               ...         ...          ...   
8597               Y             2          112500.0    361462.5      16051.5   
8598               Y             1           99000.0    675000.0      21906.0   
8599               Y             1          261000.0    711454.5      47673.0   
8600               Y             0           94500.0    270000.0      15075.0   
8601               Y             0          225000.0    500566.5      34969.5   

      ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0     ...                 0                0                0   
1     ...                 0                0                0   
2     ...                 0                0                0   
3     ...                 0                0                0   
4     ...                 1                0                0   
...   ...               ...              ...              ...   
8597  ...                 0                0                0   
8598  ...                 0                0                0   
8599  ...                 0                0                0   
8600  ...                 0                0                0   
8601  ...                 0                0                0   

     FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                   0                        0.0                       0.0   
1                   0                        0.0                       0.0   
2                   0                        0.0                       0.0   
3                   0                        0.0                       0.0   
4                   0                        0.0                       0.0   
...               ...                        ...                       ...   
8597                0                        0.0                       0.0   
8598                0                        0.0                       0.0   
8599                0                        0.0                       0.0   
8600                0                        0.0                       0.0   
8601                0                        0.0                       0.0   

      AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                            0.0                        0.0   
1                            0.0                        0.0   
2                            0.0                        1.0   
3                            0.0                        0.0   
4                            0.0                        6.0   
...                          ...                        ...   
8597     

In [29]:
#First we check null values
dataset.isnull().sum()
dataset.isna().sum()

SK_ID_CURR                    0
TARGET                        0
NAME_CONTRACT_TYPE            0
CODE_GENDER                   0
FLAG_OWN_CAR                  0
                             ..
AMT_REQ_CREDIT_BUREAU_DAY     0
AMT_REQ_CREDIT_BUREAU_WEEK    0
AMT_REQ_CREDIT_BUREAU_MON     0
AMT_REQ_CREDIT_BUREAU_QRT     0
AMT_REQ_CREDIT_BUREAU_YEAR    0
Length: 122, dtype: int64

In [30]:
dataset.shape


(8602, 122)

In [31]:
dataset=pd.read_csv("preprocessed_data.csv")
dataset

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0         100083       0         Cash loans           M            Y   
1         100145       0         Cash loans           F            Y   
2         100179       0         Cash loans           F            Y   
3         100190       0         Cash loans           M            Y   
4         100295       1         Cash loans           M            Y   
...          ...     ...                ...         ...          ...   
8597      456083       0         Cash loans           F            Y   
8598      456084       0         Cash loans           F            Y   
8599      456140       1         Cash loans           F            Y   
8600      456195       0         Cash loans           F            Y   
8601      456226       0         Cash loans           F            Y   

     FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                  Y             0          103500.0    573628.5      24435.0   
1                  Y             1          202500.0    260725.5      16789.5   
2                  N             0          202500.0    675000.0      53329.5   
3                  N             0          162000.0    263686.5      24781.5   
4                  N             1          225000.0   1019205.0      31032.0   
...              ...           ...               ...         ...          ...   
8597               Y             2          112500.0    361462.5      16051.5   
8598               Y             1           99000.0    675000.0      21906.0   
8599               Y             1          261000.0    711454.5      47673.0   
8600               Y             0           94500.0    270000.0      15075.0   
8601               Y             0          225000.0    500566.5      34969.5   

      ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0     ...                 0                0                0   
1     ...                 0                0                0   
2     ...                 0                0                0   
3     ...                 0                0                0   
4     ...                 1                0                0   
...   ...               ...              ...              ...   
8597  ...                 0                0                0   
8598  ...                 0                0                0   
8599  ...                 0                0                0   
8600  ...                 0                0                0   
8601  ...                 0                0                0   

     FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                   0                        0.0                       0.0   
1                   0                        0.0                       0.0   
2                   0                        0.0                       0.0   
3                   0                        0.0                       0.0   
4                   0                        0.0                       0.0   
...               ...                        ...                       ...   
8597                0                        0.0                       0.0   
8598                0                        0.0                       0.0   
8599                0                        0.0                       0.0   
8600                0                        0.0                       0.0   
8601                0                        0.0                       0.0   

      AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                            0.0                        0.0   
1                            0.0                        0.0   
2                            0.0                        1.0   
3                            0.0                        0.0   
4                            0.0                        6.0   
...                          ...                        ...   
8597     

In [32]:
dataset['CODE_GENDER'].value_counts()['M']

4381

In [33]:
dataset['CODE_GENDER'].value_counts()['F']

4221

In [34]:
dataset['NAME_CONTRACT_TYPE'].value_counts()['Cash loans']

7660

In [35]:
df= dataset[dataset['CODE_GENDER'] == 'M']
df

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0         100083       0         Cash loans           M            Y   
3         100190       0         Cash loans           M            Y   
4         100295       1         Cash loans           M            Y   
5         100341       0         Cash loans           M            Y   
6         100343       0         Cash loans           M            Y   
...          ...     ...                ...         ...          ...   
8585      455628       0         Cash loans           M            Y   
8589      455687       0         Cash loans           M            Y   
8591      455724       0         Cash loans           M            Y   
8593      455875       0         Cash loans           M            Y   
8596      455923       0         Cash loans           M            Y   

     FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                  Y             0          103500.0    573628.5      24435.0   
3                  N             0          162000.0    263686.5      24781.5   
4                  N             1          225000.0   1019205.0      31032.0   
5                  Y             0           76500.0    545040.0      20677.5   
6                  Y             0          315000.0     90000.0       4504.5   
...              ...           ...               ...         ...          ...   
8585               Y             1          261000.0    714154.5      38871.0   
8589               Y             0          112500.0    808650.0      26217.0   
8591               Y             1          202500.0   1350000.0      39474.0   
8593               Y             0          180000.0    735579.0      40032.0   
8596               N             0          157500.0    225000.0      15034.5   

      ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0     ...                 0                0                0   
3     ...                 0                0                0   
4     ...                 1                0                0   
5     ...                 0                0                0   
6     ...                 0                0                0   
...   ...               ...              ...              ...   
8585  ...                 0                0                0   
8589  ...                 0                0                0   
8591  ...                 0                0                0   
8593  ...                 0                0                0   
8596  ...                 0                0                0   

     FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                   0                        0.0                       0.0   
3                   0                        0.0                       0.0   
4                   0                        0.0                       0.0   
5                   0                        0.0                       0.0   
6                   0                        0.0                       0.0   
...               ...                        ...                       ...   
8585                0                        0.0                       0.0   
8589                0                        0.0                       0.0   
8591                0                        0.0                       0.0   
8593                0                        0.0                       0.0   
8596                0                        0.0                       0.0   

      AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                            0.0                        0.0   
3                            0.0                        0.0   
4                            0.0                        6.0   
5                            0.0                        0.0   
6                            0.0                        0.0   
...                          ...                        ...   
8585     

In [36]:
dataset.columns

Index(['SK_ID_CURR', 'TARGET', 'NAME_CONTRACT_TYPE', 'CODE_GENDER',
       'FLAG_OWN_CAR', 'FLAG_OWN_REALTY', 'CNT_CHILDREN', 'AMT_INCOME_TOTAL',
       'AMT_CREDIT', 'AMT_ANNUITY',
       ...
       'FLAG_DOCUMENT_18', 'FLAG_DOCUMENT_19', 'FLAG_DOCUMENT_20',
       'FLAG_DOCUMENT_21', 'AMT_REQ_CREDIT_BUREAU_HOUR',
       'AMT_REQ_CREDIT_BUREAU_DAY', 'AMT_REQ_CREDIT_BUREAU_WEEK',
       'AMT_REQ_CREDIT_BUREAU_MON', 'AMT_REQ_CREDIT_BUREAU_QRT',
       'AMT_REQ_CREDIT_BUREAU_YEAR'],
      dtype='object', length=122)

In [37]:
#1

# How many target value present in application?

In [38]:
fig = px.histogram(dataset, x='TARGET', color='TARGET',
 title= 'No. of application by Target',
 labels={'TARGET':'Target'})
fig.show()

# What are the percentage of amount income  by contract  Type in application?

In [39]:
dataset = dataset['NAME_CONTRACT_TYPE'].value_counts().reset_index()
dataset.rename(columns={'index':'NAME_CONTRACT_TYPE','NAME_CONTRACT_TYPE':'AMT_INCOME_TOTAL'}, inplace=True)
dataset

NAME_CONTRACT_TYPE  AMT_INCOME_TOTAL
0         Cash loans              7660
1    Revolving loans               942

In [40]:
fig = px.pie(dataset, names='NAME_CONTRACT_TYPE', values='AMT_INCOME_TOTAL', color='NAME_CONTRACT_TYPE',
 title='Percentage of amount income by contract Type').update_traces(textposition='inside', textinfo='percent+label')
fig.show()


# What kind of relation between AMT_ANNUITY and AMT_INCOME_TOTAL

# Correlation is Relationship between two columns.In the two column is salary column and mba_p column. It has Positive correlation,41% directly proportional.



# Which NAME_CONTRACT_TYPE	 is getting minimum AMT_INCOME_TOTAL?

In [46]:
dataset.corr()

AMT_INCOME_TOTAL
AMT_INCOME_TOTAL               1.0

In [43]:
dataset.sort_values(by="AMT_INCOME_TOTAL",ascending =True)

NAME_CONTRACT_TYPE  AMT_INCOME_TOTAL
1    Revolving loans               942
0         Cash loans              7660

# Revolving loans is getting minimum AMT_INCOME_TOTAL .The minimum AMT_INCOME_TOTAL  is 942

# Test the similiarity between gender(male) with respect to target and amt income total 

In [63]:
# from scipy.stats import ttest_ind
#drop null values
dataset=dataset.dropna()
male=dataset[dataset["CODE_GENDER"]=="M"]["TARGET"]
male1=dataset[dataset["CODE_GENDER"]=="M"]["AMT_INCOME_TOTAL"]
#print(male)
ttest_ind(male,male1)

Ttest_indResult(statistic=-103.93546356802855, pvalue=0.0)

In [52]:
#!pip install statsmodels

In [54]:
dataset=pd.read_csv("preprocessed_data.csv")
dataset

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0         100083       0         Cash loans           M            Y   
1         100145       0         Cash loans           F            Y   
2         100179       0         Cash loans           F            Y   
3         100190       0         Cash loans           M            Y   
4         100295       1         Cash loans           M            Y   
...          ...     ...                ...         ...          ...   
8597      456083       0         Cash loans           F            Y   
8598      456084       0         Cash loans           F            Y   
8599      456140       1         Cash loans           F            Y   
8600      456195       0         Cash loans           F            Y   
8601      456226       0         Cash loans           F            Y   

     FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                  Y             0          103500.0    573628.5      24435.0   
1                  Y             1          202500.0    260725.5      16789.5   
2                  N             0          202500.0    675000.0      53329.5   
3                  N             0          162000.0    263686.5      24781.5   
4                  N             1          225000.0   1019205.0      31032.0   
...              ...           ...               ...         ...          ...   
8597               Y             2          112500.0    361462.5      16051.5   
8598               Y             1           99000.0    675000.0      21906.0   
8599               Y             1          261000.0    711454.5      47673.0   
8600               Y             0           94500.0    270000.0      15075.0   
8601               Y             0          225000.0    500566.5      34969.5   

      ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0     ...                 0                0                0   
1     ...                 0                0                0   
2     ...                 0                0                0   
3     ...                 0                0                0   
4     ...                 1                0                0   
...   ...               ...              ...              ...   
8597  ...                 0                0                0   
8598  ...                 0                0                0   
8599  ...                 0                0                0   
8600  ...                 0                0                0   
8601  ...                 0                0                0   

     FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                   0                        0.0                       0.0   
1                   0                        0.0                       0.0   
2                   0                        0.0                       0.0   
3                   0                        0.0                       0.0   
4                   0                        0.0                       0.0   
...               ...                        ...                       ...   
8597                0                        0.0                       0.0   
8598                0                        0.0                       0.0   
8599                0                        0.0                       0.0   
8600                0                        0.0                       0.0   
8601                0                        0.0                       0.0   

      AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                            0.0                        0.0   
1                            0.0                        0.0   
2                            0.0                        1.0   
3                            0.0                        0.0   
4                            0.0                        6.0   
...                          ...                        ...   
8597     

In [55]:
dataset[dataset["CODE_GENDER"]=="M"]

SK_ID_CURR  TARGET NAME_CONTRACT_TYPE CODE_GENDER FLAG_OWN_CAR  \
0         100083       0         Cash loans           M            Y   
3         100190       0         Cash loans           M            Y   
4         100295       1         Cash loans           M            Y   
5         100341       0         Cash loans           M            Y   
6         100343       0         Cash loans           M            Y   
...          ...     ...                ...         ...          ...   
8585      455628       0         Cash loans           M            Y   
8589      455687       0         Cash loans           M            Y   
8591      455724       0         Cash loans           M            Y   
8593      455875       0         Cash loans           M            Y   
8596      455923       0         Cash loans           M            Y   

     FLAG_OWN_REALTY  CNT_CHILDREN  AMT_INCOME_TOTAL  AMT_CREDIT  AMT_ANNUITY  \
0                  Y             0          103500.0    573628.5      24435.0   
3                  N             0          162000.0    263686.5      24781.5   
4                  N             1          225000.0   1019205.0      31032.0   
5                  Y             0           76500.0    545040.0      20677.5   
6                  Y             0          315000.0     90000.0       4504.5   
...              ...           ...               ...         ...          ...   
8585               Y             1          261000.0    714154.5      38871.0   
8589               Y             0          112500.0    808650.0      26217.0   
8591               Y             1          202500.0   1350000.0      39474.0   
8593               Y             0          180000.0    735579.0      40032.0   
8596               N             0          157500.0    225000.0      15034.5   

      ...  FLAG_DOCUMENT_18 FLAG_DOCUMENT_19 FLAG_DOCUMENT_20  \
0     ...                 0                0                0   
3     ...                 0                0                0   
4     ...                 1                0                0   
5     ...                 0                0                0   
6     ...                 0                0                0   
...   ...               ...              ...              ...   
8585  ...                 0                0                0   
8589  ...                 0                0                0   
8591  ...                 0                0                0   
8593  ...                 0                0                0   
8596  ...                 0                0                0   

     FLAG_DOCUMENT_21 AMT_REQ_CREDIT_BUREAU_HOUR AMT_REQ_CREDIT_BUREAU_DAY  \
0                   0                        0.0                       0.0   
3                   0                        0.0                       0.0   
4                   0                        0.0                       0.0   
5                   0                        0.0                       0.0   
6                   0                        0.0                       0.0   
...               ...                        ...                       ...   
8585                0                        0.0                       0.0   
8589                0                        0.0                       0.0   
8591                0                        0.0                       0.0   
8593                0                        0.0                       0.0   
8596                0                        0.0                       0.0   

      AMT_REQ_CREDIT_BUREAU_WEEK  AMT_REQ_CREDIT_BUREAU_MON  \
0                            0.0                        0.0   
3                            0.0                        0.0   
4                            0.0                        6.0   
5                            0.0                        0.0   
6                            0.0                        0.0   
...                          ...                        ...   
8585     

# Test the similarity between the male and female Target

In [56]:
from scipy.stats import ttest_ind
#drop null values
dataset=dataset.dropna()
male=dataset[dataset["CODE_GENDER"]=="M"]["TARGET"]
female=dataset[dataset["CODE_GENDER"]=="F"]["TARGET"]
#print(male)
ttest_ind(male,female)

Ttest_indResult(statistic=3.071560910453999, pvalue=0.002136078503149578)

# Test the similarity between the male and female amount income total

In [57]:
from scipy.stats import ttest_ind
#drop null values
dataset=dataset.dropna()
male=dataset[dataset["CODE_GENDER"]=="M"]["AMT_INCOME_TOTAL"]
female=dataset[dataset["CODE_GENDER"]=="F"]["AMT_INCOME_TOTAL"]
#print(male)
ttest_ind(male,female)

Ttest_indResult(statistic=9.774491475186364, pvalue=1.8958257274241107e-22)

# Test the similarity between the CODE_GENDER  with respect to TARGET  and AMT_INCOME_TOTAL.

In [58]:
from scipy.stats import ttest_ind
#drop null values
dataset=dataset.dropna()
male=dataset[dataset["CODE_GENDER"]=="M"]["TARGET"]
male1=dataset[dataset["CODE_GENDER"]=="F"]["AMT_INCOME_TOTAL"]
#print(male)
ttest_ind(male,male1)

Ttest_indResult(statistic=-106.7789197894265, pvalue=0.0)

# Test the Analysis of variance between AMT_INCOME_TOTAL , AMT_CREDIT and AMT_ANNUITY  

In [59]:
import scipy.stats as stats
stats.f_oneway(dataset["AMT_INCOME_TOTAL"],dataset["AMT_CREDIT"],dataset["AMT_ANNUITY"])

F_onewayResult(statistic=12935.897446153398, pvalue=0.0)

# Find the two-way ANOVA

In [60]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
# Two-way ANOVA
ano=ols("AMT_INCOME_TOTAL~TARGET+AMT_CREDIT",data=dataset).fit()
two=sm.stats.anova_lm(ano,type=2)

In [61]:
two

df        sum_sq       mean_sq            F         PR(>F)
TARGET         1.0  2.719577e+11  2.719577e+11    15.496412   8.331383e-05
AMT_CREDIT     1.0  2.083568e+13  2.083568e+13  1187.237258  7.924236e-244
Residual    8599.0  1.509101e+14  1.754972e+10          NaN            NaN